docs/fundamentals/databases.md

```diff
@@ -0,0 +1,58 @@
+| `memCompGauge`       | Gauge for tracking the number of memory compaction                      |
+| `level0CompGauge`    | Gauge for tracking the number of table compaction in level0             |
+| `nonlevel0CompGauge` | Gauge for tracking the number of table compaction in non0 level         |
+| `seekCompGauge`      | Gauge for tracking the number of table compaction caused by read opt    |
+
+
+## Freezer/ancients
+
+Older segments of the chain are moved out of the fast LevelDB database and into a freezer database the is less performant. Nodes rarely need to access these files so IO speed is less important and the bulk of the chain data can be stored on a cheaper HDD. Once blocks pass some threshold age (128 blocks behind the head by default) the block and receipt data is flattened and saved as a raw binary blob of data along with an index entry file used for identification. This flat file format allows Geth to capitalize on the operating system's built-in memory manager and file cacher which in turn allows Geth to maximize the available memory without hogging it from other processes. This is good for overall system stability and also reduces the strain on Go's garbage collector and reduces the total amount of memory used by Geth. 
+
+Geth also tracks some basic metrics relating to the ancients database that can be monitored:
+
+| metric       | function                                                   |
+| ------------ | ---------------------------------------------------------- |
+| `readMeter`  | Meter for measuring the effective amount of data read      |
+| `writeMeter` | Meter for measuring the effective amount of data written   |
+| `sizeGauge`  | Gauge for tracking the combined size of all freezer tables |
+
+
+The ancients data is saved entirely separately from the fast-access recent data, meaning it can be stored in a  different location. The default location for the ancient chain segments is inside the `chaindata` directory, which is inside `datadir`, but it can be defined by passing `--datadir.ancient <path>` to Geth on startup. This path can be to a slow HDD disk, permitting people to run Ethereum nodes without requiring a huge SSD. The ancient can also be moved later by manually copying the directory to a new location and then starting Geth passing the new path to `--datadir.ancient`.
+
+
+## Using the freezer
+
+If Geth cannot find the freezer, either because the database was deleted or because Geth received an incorrect path, then the node becomes unstable. It is explicitly forbidden to start Geth with an invalid path to the freezer. However, if the state database goes missing it can be rebuilt "on-the-fly" by Geth. In this case, Geth rebuilds all its indices based upon data from the freezer and then does a snap-sync on top of it to rebuild the missing state data. In summary - Geth can rebuild the recent past from the ancient past but not vice-versa.
```

Comment:
> However, if the state database goes missing it can be rebuilt "on-the-fly" by Geth.

Wait a minute... How do you mean? You must be thinking of a different scenario than I am, because if you nuke the statedb and keep only ancients, this is not at all possible.

Reply:
Why? If we plugin the existent ancient into a empty Geth instance, the ancient can be reused and run snap-sync on top.

Reply:
will wait for clarification from you guys before updating this - my understanding is that losing the state db could be recovered by snap-syncing on top of the ancients. This is also how it is described here in the "Freezer Tricks" section: https://blog.ethereum.org/2019/07/10/geth-v1-9-0.

Reply:
Or rather, you compared two scenarios, and it sounded like you need to be careful with ancients, geth can become unstable if you mess with them. But state can be fixed "on the fly" if you mess it up. 

And that feels a bit odd, to me

Reply:
ok, maybe just delete the last sentence then? It's kind of unnecessary and might have muddied the waters.

```suggestion
If Geth cannot find the freezer, either because the database was deleted or because Geth received an incorrect path, then the node becomes unstable. It is explicitly forbidden to start Geth with an invalid path to the freezer. However, if the state database goes missing it can be rebuilt "on-the-fly" by Geth. In this case, Geth rebuilds all its indices based upon data from the freezer and then does a snap-sync on top of it to rebuild the missing state data.
```